<?xml version="1.0"?>
<launch>
  <!-- dummy detection and sensor data -->
  <arg name="visible_range" default="300.0"/>
  <arg name="use_base_link_z" default="true"/>

  <group>
    <push-ros-namespace namespace="simulation"/>
    <node pkg="dummy_perception_publisher" exec="dummy_perception_publisher_node" name="dummy_perception_publisher" output="screen">
      <remap from="output/dynamic_object" to="/perception/object_recognition/detection/objects_with_feature"/>
      <remap from="output/objects_pose" to="debug/object_pose"/>
      <remap from="output/points_raw" to="/perception/obstacle_segmentation/pointcloud"/>
      <remap from="input/object" to="dummy_perception_publisher/object_info"/>
      <remap from="input/reset" to="input/reset"/>
      <param name="visible_range" value="$(var visible_range)"/>
      <param name="detection_successful_rate" value="1.0"/>
      <param name="enable_ray_tracing" value="false"/>
      <param name="use_object_recognition" value="false"/>
      <param name="use_base_link_z" value="$(var use_base_link_z)"/>
    </node>

    <!-- convert DynamicObjectsWithFeatureArray to DynamicObjects -->
    <!-- これPsimでは必要?? -->
    <include file="$(find-pkg-share detected_object_feature_remover)/launch/detected_object_feature_remover.launch.xml">
      <arg name="input" value="/perception/object_recognition/detection/objects_with_feature"/>
      <arg name="output" value="/perception/object_recognition/detection/objects"/>
    </include>
  </group>
</launch>
